libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -956,4 +969,54 @@ bool AP_Mount_Siyi::get_rangefinder_distance(float& distance_m) const
     return true;
 }
 
+bool AP_Mount_Siyi::_check_firmware_version() const {
+    if (!_fw_version.received) {
+        debug("Can't check firmware if we haven't received it...");
+        return false;
+    }
+
+    if (!_got_hardware_id) {
+        debug("Can't check firmware without Hardware ID!");
+        return false;
+    }
+
+    FirmwareVersion minimum_ver {};
+    switch (_hardware_model) {
+        case HardwareModel::A8:
+            minimum_ver.camera.major = 0;
+            minimum_ver.camera.minor = 2;
+            minimum_ver.camera.patch = 1;
+            break;
+
+        case HardwareModel::A2:
+        case HardwareModel::ZR10:
+        case HardwareModel::ZR30:
+        case HardwareModel::ZT30:
+            // TBD
+            break;
+
+        case HardwareModel::UNKNOWN:
+            debug("Can't check FW on unknown hardware model!");
+            return false;
+    }
+
+    const uint32_t minimum_camera_val =  (minimum_ver.camera.major << 16) + (minimum_ver.camera.minor << 8) + minimum_ver.camera.patch;
+    const uint32_t firmware_camera_val = (_fw_version.camera.major << 16) + (_fw_version.camera.minor << 8) + _fw_version.camera.patch;
+
+    const bool is_camera_supported = firmware_camera_val >= minimum_camera_val;
+
+    if (!is_camera_supported) {
+        GCS_SEND_TEXT(
+            MAV_SEVERITY_WARNING,
+            "Mount: Siyi running old camera firmware (got v%u.%u.%u, expected v%u.%u.%u)!",
+            _fw_version.camera.major, _fw_version.camera.minor, _fw_version.camera.patch,
+            minimum_ver.camera.major, minimum_ver.camera.minor, minimum_ver.camera.patch
+        );
+    } else {
+        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Mount: Siyi firmware is up-to-date");
```

Comment:
Let's remove this positive message because too many messages to the user mean they will stop reading all messages.

Reply:
Oops, sorry, was using that for debugging. Removed.